src/OpenTelemetry.Abstractions/Trace/ITracerFactory.cs

```diff
@@ -0,0 +1,35 @@
+﻿// <copyright file="ITracerFactory.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Trace
+{
+    using OpenTelemetry.Context;
+    using OpenTelemetry.Context.Propagation;
+
+    /// <summary>
+    /// Creates Tracers for an instrumentation library.
+    /// </summary>
+    public interface ITracerFactory
```

Comment:
please change it to abstract class (we'll be doing it for everything). interfaces cannot be changed without breaking changes, abstract classes at least leave a room for extending them

Reply:
OK. What's the naming convention going to be? Are we keeping the "I" prefix?

Reply:
Nope, no I prefix, just abstract TraceFactory

Reply:
So, are the namespaces going to change then?
Because with that change we would end up with the same type names in API (Abstractions) and SDK.

Reply:
I updated the PR and I left the "I" prefix in for now, until above question about clashing names is resolved.